### ISSUE TITLE
Guided Tours - back button

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#40009]
### Steps to reproduce the issue
See video recording - keyboard navigation only - all key presses highlighted on screen by external software
Creating seperate issue for each bug report so that they dont get lost inside a bigger report (again)

Using the keyboard I cannot reach the back button unless i shift tab. 

https://user-images.githubusercontent.com/1296369/222924946-1488d06d-6238-4432-88c2-602c50be7ec8.mp4

The requirement to use a mouse to successfully use the Guided Tours component is an accessibility failure and breaks our commitment to providing an accessible way to use Joomla

**Comment by obuisard** [comment]
The order in which buttons are added to a step is as follow:
cancel - back - next (in Firefox, there is a way to check tabIndex and they appear in the right order).
When on cancel (why cancel? Because this is the only available button on all steps), the script gives focus to the target, if there is one.
When in an target, the script gives back the focus to the step (using the blur event).
These are the only manipulations made. The rest is the browser deciding how to tab through.
Manipulating tabIndex is usually not a good idea, that is why we don't enforce anything more than what I explained.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40009">issues.joomla.org/tracker/joomla-cms/40009</a>.</sub>

**Comment by brianteeman** [comment]
Something is wrong then as it works correctly in the shepherd demo

**Comment by obuisard** [comment]
It also works well when a step is not interactive.
In an interactive step, this behavior comes from the fact that once we get out of focus from the target, the focus is moved to the 'Next' button, if it exists and not disabled, allowing the user to move on to the next step faster. Therefore, going 'Back' can only be done with shift-tab.
We can give focus to the 'back' button first, no problem, but does it make sense once a value has been entered in an input field? 
BTW, we always go through the cancel button, it's just the way modals behave.

**Comment by brianteeman** [comment]
addressed elsewhere
